src/api/v2/ws.rs

```diff
@@ -44,31 +48,38 @@ pub async fn connect(
 			Ok(message) if !message.is_text() => continue,
 			Ok(message) => {
 				if let Some(error) = match <Request>::try_from(message) {
-					Ok(request) => {
-						if let Err(error) = handle_websocket_message(
-							request,
-							sender.clone(),
-							&version,
-							&config,
-							&node,
-							state.clone(),
-						) {
+					Ok(request) => match handle_request(
+						request,
+						&version,
+						&config,
+						&node,
+						submitter.clone(),
+						state.clone(),
+					)
+					.await
+					{
+						Ok(message) => {
+							if let Err(error) = send_response(sender.clone(), message) {
+								error!("Error handling web socket message: {error}");
+								Some(Error::internal_server_error())
+							} else {
+								None
+							}
+						},
+						Err(error) => {
 							error!("Error handling web socket message: {error}");
 							Some(Error::internal_server_error())
-						} else {
-							None
-						}
+						},
 					},
 					Err(error) => {
 						warn!("Error handling web socket message: {error}");
-						Some(Error::bad_request(
-							"Error handling web socket message: Failed to parse request"
-								.to_string(),
-						))
+						Some(Error::bad_request_unknown(&format!(
+							"Error handling web socket message: Failed to parse request: {error}"
+						)))
 					},
 				} {
-					if let Err(error) = sender.send(Ok(Message::text::<String>(error.into()))) {
-						error!("{error}");
+					if let Err(error) = send_response(sender.clone(), error.into()) {
+						error!("Failed to send error: {error}");
 					};
 				};
```

Comment:
It became a bit hard to follow this. It is better to remove nesting like this:
```rust
let message = match result {
	Err(error) =>error!("Error receiving client message: {error}"),
    Ok(message) if !message.is_text() => continue,
    Ok(message) => message,
};
let request = match Request::try_from(message) {
    Err(error) => {
    	warn!("Error handling web socket message: {error}");
		let error = Error::bad_request_unknown(&format!(
			"Error handling web socket message: Failed to parse request: {error}"
		));
		if let Err(error) = send_response(sender.clone(), error.into()).await {
			error!("Failed to send error: {error}");
		}
		continue;
    },
	Ok(request) => request,
};

let response = match handle_request(
	request,
	&version,
	&config,
	&node,
	submitter.clone(),
	state.clone(),
)
.await
{
	Ok(response) => response,
	Err(error) => {
		error!("Error handling web socket message: {error}");
		Error::internal_server_error().into()
	}
};

if let Err(error) = send_response(sender.clone(), response) {
	error!("Failed to send response: {error}");
}
```

Reply:
Sure, its a complext flow, its more clear now.

Reply:
Thanks!